internal/compiler/layout.rs

```diff
@@ -315,28 +315,53 @@ impl Padding {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct Spacing {
+    pub horizontal: Option<NamedReference>,
+    pub vertical: Option<NamedReference>,
+}
+
+impl Spacing {
+    fn visit_named_references(&mut self, visitor: &mut impl FnMut(&mut NamedReference)) {
+        if let Some(e) = self.horizontal.as_mut() {
+            visitor(&mut *e);
+        }
+        if let Some(e) = self.vertical.as_mut() {
+            visitor(&mut *e);
+        }
+    }
+
+    pub fn orientation(&self, o: Orientation) -> Option<&NamedReference> {
+        match o {
+            Orientation::Horizontal => self.horizontal.as_ref(),
+            Orientation::Vertical => self.vertical.as_ref(),
+        }
+    }
+}
+
 #[derive(Debug, Clone)]
 pub struct LayoutGeometry {
     pub rect: LayoutRect,
-    pub spacing: Option<NamedReference>,
+    pub spacing: Spacing,
     pub alignment: Option<NamedReference>,
     pub padding: Padding,
 }
 
 impl LayoutGeometry {
     pub fn visit_named_references(&mut self, visitor: &mut impl FnMut(&mut NamedReference)) {
         self.rect.visit_named_references(visitor);
-        if let Some(e) = self.spacing.as_mut() {
-            visitor(&mut *e)
-        }
         if let Some(e) = self.alignment.as_mut() {
             visitor(&mut *e)
         }
+        self.spacing.visit_named_references(visitor);
         self.padding.visit_named_references(visitor);
     }
 
     pub fn new(layout_element: &ElementRc) -> Self {
-        let spacing = binding_reference(layout_element, "spacing");
+        let spacing = || binding_reference(layout_element, "spacing");
+        init_fake_property(layout_element, "spacing-horizontal", spacing);
+        init_fake_property(layout_element, "spacing-vertical", spacing);
```

Comment:
I think it's fine if this code stays there also for Horizontal/VerticalLayout,  because these property can't exist on such layout. 
But there could be a test that this is indeed an error.

Reply:
I added a syntax test for this.